### ISSUE TITLE
ReferenceError: TextEncoder is not defined

### ISSUE THREAD

**Issue by commonuserlol** [paulmillr/noble-ciphers#31]
Hey, i want to use this library in embedded environment where no `TextEncoder`, can you switch to pure js way for this?

**Comment by paulmillr** [comment]
https://gist.github.com/Yaffle/5458286

**Comment by paulmillr** [comment]
we have no hard dependency on textencoder though

**Comment by farwayer** [comment]
Unfortunately the `utf8ToBytes` function is called at the top level of the `_arx.ts` module, which causes an error in environments where `TextEncoder` is not supported (React Native for example)
```js
const sigma16 = utf8ToBytes('expand 16-byte k');
const sigma32 = utf8ToBytes('expand 32-byte k');
```

**Comment by paulmillr** [comment]
fixed in e1046f5, will release new release soon
